ecl/resource_ecl_provider_connectivity_tenant_connection_v2.go

```diff
@@ -0,0 +1,461 @@
+		}
+		attachmentOpts = serverAttachmentOpts
+	}
+
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	opts := tenant_connections.CreateOpts{
+		Name:                      d.Get("name").(string),
+		Description:               d.Get("description").(string),
+		Tags:                      getTags(d, "tags"),
+		TenantConnectionRequestID: d.Get("tenant_connection_request_id").(string),
+		DeviceType:                deviceType,
+		DeviceID:                  d.Get("device_id").(string),
+		DeviceInterfaceID:         deviceInterfaceId,
+		AttachmentOpts:            attachmentOpts,
+	}
+	log.Printf("[DEBUG] Create Options: %#v", opts)
+
+	tenantConnection, err := tenant_connections.Create(client, opts).Extract()
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity Tenant Connection: %s", err)
+	}
+
+	d.SetId(tenantConnection.ID)
+	log.Printf("[DEBUG] Created ECL Provider Connectivity Tenant Connection %s: %#v", tenantConnection.Name, tenantConnection)
+	log.Printf("[DEBUG] Waiting for Tenant Connection (%s) to become active", tenantConnection.ID)
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"creating"},
+		Target:     []string{"active"},
+		Refresh:    waitForTenantConnectionActive(client, tenantConnection.ID),
+		Timeout:    d.Timeout(schema.TimeoutCreate),
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	if deviceType == "ECL::VirtualNetworkAppliance::VSRX" {
+		stateConf.PollInterval = vnaCreatePollInterval
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		return fmt.Errorf(
+			"Error waiting for Tenant Connection (%s) to become active: %s",
+			tenantConnection.ID, err)
+	}
+
+	return resourceProviderConnectivityTenantConnectionV2Read(d, meta)
+}
+
+func resourceProviderConnectivityTenantConnectionV2Read(d *schema.ResourceData, meta interface{}) error {
+	config := meta.(*Config)
+	client, err := config.providerConnectivityV2Client(GetRegion(d, config))
+	if err != nil {
+		return fmt.Errorf("error creating ECL Provider Connectivity client: %s", err)
+	}
+
+	tenantConnection, err := tenant_connections.Get(client, d.Id()).Extract()
+	if err != nil {
+		return CheckDeleted(d, err, "Provider Connectivity Tenant Connection")
+	}
+	log.Printf("[DEBUG] Retrieved Provider Connectivity Tenant Connection %s: %+v", tenantConnection.Name, tenantConnection)
+
+	d.SetId(tenantConnection.ID)
+	d.Set("tenant_connection_request_id", tenantConnection.TenantConnectionRequestID)
+	d.Set("name", tenantConnection.Name)
+	d.Set("description", tenantConnection.Description)
+	d.Set("tags", tenantConnection.Tags)
+	d.Set("tenant_id", tenantConnection.TenantID)
+	d.Set("name_other", tenantConnection.NameOther)
+	d.Set("description_other", tenantConnection.DescriptionOther)
+	d.Set("tags_other", tenantConnection.TagsOther)
```

Comment:
Can you handle this properly? I mean are you sure that any mismatch never happen between state and template?
https://github.com/nttcom/eclcloud/pull/21#discussion_r375617661

Reply:
Yes, confirm.